File: typescript/src/structural/bridge/Bridge.ts
// Interfaz
interface Transmission {
    transmit(data: string): void;
}
  
// Implementaciones concretas
class Bluetooth implements Transmission {
    transmit(data: string): void {
        console.log(`🔵 Sending via Bluetooth: ${data}`);
    }
}
  
class Cable implements Transmission {
    transmit(data: string): void {
        console.log(`🔌 Sending via Cable: ${data}`);
    }
}
  
class Wifi implements Transmission {
    transmit(data: string): void {
        console.log(`📡 Sending via Wifi: ${data}`);
    }
}
  
  // Abstracción
class AudioDevice {
    protected transmission: Transmission;
  
    constructor(transmission: Transmission) {
        this.transmission = transmission;
    }
  
    protected transmit(data: string): void {
        this.transmission.transmit(data);
    }
}
  
  // Dispositivos concretos
class HeadPhones extends AudioDevice {
    playSound(sound: string): void {
        console.log("🎧 Headphones playing...");
        this.transmit(sound);
    }
}
  
class Speaker extends AudioDevice {
    playSound(sound: string): void {
        console.log("🔊 Speaker playing...");
        this.transmit(sound);
    }
}
  
export { Bluetooth, Wifi, HeadPhones, Speaker };
  

File: python/structural/bridge/bridge.py
from abc import ABC, abstractmethod

# Interfaz esperada 
class Transmission(ABC):
    @abstractmethod
    def transmit(self, data: str) -> None:
        pass

# Implementaciones concretas
class Bluetooth(Transmission):
    def transmit(self, data: str) -> None:
        print(f"🔵 Sending via Bluetooth: {data}")

class Cable(Transmission):
    def transmit(self, data: str) -> None:
        print(f"🔌 Sending via Cable: {data}")

class Wifi(Transmission):
    def transmit(self, data: str) -> None:
        print(f"📡 Sending via Wifi: {data}")

# Abstracción
class AudioDevice(ABC):
    def __init__(self, transmission: Transmission):
        self.transmission = transmission

    @abstractmethod
    def play_sound(self, sound: str) -> None:
        pass

# Dispositivos concretos
class Headphones(AudioDevice):
    def play_sound(self, sound: str) -> None:
        print("🎧 Headphones playing...")
        self.transmission.transmit(sound)

class Speaker(AudioDevice):
    def play_sound(self, sound: str) -> None:
        print("🔊 Speaker playing...")
        self.transmission.transmit(sound)

